abixen-platform-core/src/test/java/com/abixen/platform/core/infrastructure/configuration/PlatformTestUserDetailsService.java

```diff
@@ -0,0 +1,103 @@
+import com.abixen.platform.core.application.service.UserManagementService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+
+public class PlatformTestUserDetailsService implements UserDetailsService {
+
+    private final Logger log = LoggerFactory.getLogger(PlatformTestUserDetailsService.class.getName());
+
+    @Autowired
+    private UserManagementService userManagementService;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        log.debug("loadUserByUsername({})", username);
+
+        final UserDto user = userManagementService.findUser(username);
+        log.debug("user: {}", user);
+
+
+        if (user == null) {
+            log.error("Wrong username: {}", username);
+            throw new UsernameNotFoundException("Wrong username and / or password.");
+        }
+
+        if (!UserState.ACTIVE.equals(user.getState())) {
+            log.error("User is inactive: {}", username);
+            throw new UsernameNotFoundException("User is inactive.");
+        }
+
+        boolean enabled = true;
+        boolean accountNonExpired = true;
+        boolean credentialsNonExpired = true;
+        boolean accountNonLocked = true;
+
+        final Collection<? extends GrantedAuthority> authorities = getAuthorities(user);
+        final boolean admin = isAdmin(authorities);
+
+        final PlatformUser platformUser = new PlatformUser(user.getUsername(),
+                user.getPassword(),
+                enabled,
+                accountNonExpired,
+                credentialsNonExpired,
+                accountNonLocked,
+                authorities,
+                user.getFirstName(),
+                user.getLastName(),
+                admin,
+                user.getId(),
+                user.getSelectedLanguage(),
+                user.getAvatarFileName());
+
+        return platformUser;
+    }
+
+    private boolean isAdmin(Collection<? extends GrantedAuthority> authorities) {
+        for (GrantedAuthority grantedAuthority : authorities) {
+            if ("ROLE_ADMIN".equals(grantedAuthority.getAuthority())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Collection<? extends GrantedAuthority> getAuthorities(UserDto user) {
+        List<GrantedAuthority> grantedAuthorities = new ArrayList<GrantedAuthority>();
+        for (RoleDto role : user.getRoles()) {
```

Comment:
Lambda expression will be better. 
return user.getRoles().stream
.filter()
,map()
.collect()